apis/nucleus/src/hooks/useSessionModel.js

```diff
@@ -19,10 +29,11 @@ export default function useSessionModel(definition, app, ...deps) {
     }
     // Create new session object
     const create = async () => {
-      if (!rpcShared) {
-        const rpc = app.createSessionObject(definition);
-        rpcShared = rpc;
+      if (allocatedResolve) {
+        rpcShared = app.createSessionObject(definition);
         rpcRequestSessionModelStore.set(key, rpcShared);
+        const resolvedModel = await rpcShared;
+        allocatedResolve(resolvedModel);
```

Comment:
I think it would be a better solution to move `rpcRequestSessionModelStore.get(key);` inside the create function instead

if not you should set allocatedResolve to undefined/null when used to not create extra session objects if create run again for an old object

Reply:
yea, that is more appropriate and simpler, good